File: src/main/java/io/funky/fangs/keep_it_personal/command/ViewDeathPreferencesCommand.java
package io.funky.fangs.keep_it_personal.command;

import com.mojang.brigadier.context.CommandContext;
import io.funky.fangs.keep_it_personal.domain.DeathPreference;
import io.funky.fangs.keep_it_personal.domain.DeathPreferenceContainer;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.network.chat.Component;

import static java.util.stream.Collectors.joining;

public class ViewDeathPreferencesCommand extends KeepItPersonalCommand {
    private static final String MESSAGE_TEMPLATE = "Keeping: [%s]";

    @Override
    public Integer run(CommandContext<CommandSourceStack> context, DeathPreferenceContainer container) {
        final var preferences = container.getDeathPreferences();
        context.getSource().sendSuccess(() -> Component.literal(MESSAGE_TEMPLATE.formatted(
                preferences.stream()
                        .map(DeathPreference::toString)
                        .collect(joining(", "))
        )), false);
        return SINGLE_SUCCESS;
    }
}

File: src/main/kotlin/io/funky/fangs/keep_it_personal/command/ViewDeathPreferencesCommand.kt
package io.funky.fangs.keep_it_personal.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.text.Text

class ViewDeathPreferencesCommand: KeepItPersonalCommand() {
    override fun run(
        keepCommandState: KeepCommandState,
        context: CommandContext<ServerCommandSource>,
        player: ServerPlayerEntity
    ): Int {
        val preferences = keepCommandState.getPlayerPreferences(player.uuid)
        val message = Text.literal(
            "Keeping: [${preferences.joinToString(", ")}]"
        )
        context.source.sendFeedback({ message }, false)
        return Command.SINGLE_SUCCESS
    }
}
